### DISCUSSION TITLE
Carbon Language community transparency report through 2023-01-25

### DISCUSSION THREAD

**Discussion by CelineausBerlin** [carbon-language/carbon-lang#2556] [Transparency reports]
The Carbon community works to be welcoming and kind among itself and to others, with a deep commitment to psychological safety, and we want to ensure that doesn’t change as we grow and evolve. To that end, we have a few ground rules that we ask all community members to adhere to:

-   be welcoming,
-   be friendly and patient,
-   be considerate,
-   be kind,
-   be careful in the words that we choose, when we disagree, we try to understand why, and recognize when progress has stopped, and take a step back.

The following summary is intended to help the community understand what kinds of Code of Conduct (CoC) incidents were brought to our attention lately, and how we dealt with them.

Publishing such transparency reports on regular basis is helping us track progress and hold ourselves accountable to high standards of community culture.

### Summary

Since September 2022, we updated our AutoMod bots that are automatically catching the use of harmful language, and we added an interaction limit for new users on GitHub to prevent spam for 7 days. Our AutoMod bots have blocked 42 messages. There have been 27 other documented incidents during that same period. That means 69+ conduct-related incidents were taken care of by our moderation team, in one way or another.

Please note that some incidents may have escaped our attention. Please help us keep our spaces welcoming and fostering a spirit of collaboration, and report any situation that may require our intervention: https://github.com/carbon-language/carbon-lang/blob/trunk/CODE_OF_CONDUCT.md

### Our interventions from _2022-09-01_ through _2022-12-31_

The Carbon moderation team, including our Code of Conduct team, is getting the support of an external professional community culture and conduct expert.

The moderation team is also running an internal book club to expand their knowledge. 
Currently, the book club is learning more about anti-racist practices.

These are the conduct incidents that were brought to our attention between September and December 2022:

-   > _"**Be welcoming.** We strive to be a community that welcomes and supports people of all backgrounds and identities. This includes, but is not limited to, members of any race, ethnicity, culture, national origin, color, immigration status, social and economic class, educational level, sex, sexual orientation, gender identity and expression, physical appearance, age, size, family status, relationship status, political belief, religion or lack thereof, and mental and physical ability."
    > Carbon community Code of Conduct, version from 2022-10-20_

    The Carbon community has been addressed in the default masculine on Discord 30+ times, resulting in messages being automatically blocked. We are now sending a message about these blocks so users get the chance to understand why their message was blocked, which was not apparent to many. There were also 13 documented cases of the use of otherwise harmful language.

    Combined, these resulted in educational opportunities (3), message editing/hiding (32), and warnings (3). 42 cases happened on Discord, 2 on GitHub.

-   > _“**Be kind.** Not all of us will agree all the time, but disagreement is no excuse for poor behavior and hurtful words. We might all experience some frustration now and then, but we cannot allow that frustration to turn into a personal attack. It’s important to remember that a community where people feel threatened is not a productive one. Members of our community should be 
kind when dealing with other members as well as with people outside the Carbon community.”
    > Carbon community Code of Conduct, version from 2022-10-20_

    There were 4 cases of destructive comments, 2 on Discord and 2 on GitHub. 2 users got a warning, 1 situation was used as an opportunity to explain our policies, and we started an on-going conversation with one person. Our moderation team intervened in 6 trolling situations, 4 on Discord and 2 on GitHub. 6 users involved in trolling behavior were banned.

-   > _“**Recognize when progress has stopped, and take a step back.** Regardless of whether you're trying to resolve a disagreement or anything else, think about whether you're making progress. Sometimes messaging doesn't give time to think about a situation fully, and repeating positions can make people defensive. Step back for a few minutes or hours to think through the issue before responding again. Consider pulling in another community member to give a fresh perspective. Maybe meet over VC instead. Switching approaches can help resume progress.”
    > Carbon community Code of Conduct, version from 2022-10-20_

    12 conversations were deemed off-topic, 5 on Discord and 7 on GitHub. 2 issues were closed, 7 posts were hidden or deleted, 6 users were warned, and 3 banned

-   **Other interventions**

    Our CoC team helped a person on Twitter understand our Code of Conduct after they complained about our safety approach.

    We closed a GitHub discussion in which a logo was posted, in agreement with the person who posted it.

### Closing observations

The intensity of this work has been moderate in the last months. We’ll keep watching and providing support when needed.

Thank you for your contributions and see you around,
-- The (interim) Carbon Language conduct team: our community lead, together with the Carbon leads

**Comment by ki37ki** [comment]
Great👍👍👍

**Comment by prateekthakur272** [comment]
Sounds good😃
